ntoskrnl/ke/amd64/trap.S

```diff
@@ -966,6 +975,28 @@ FUNC KiZwSystemService
 
 ENDFUNC
 
+PUBLIC KiConvertToGuiThread
+FUNC KiConvertToGuiThread
+
+    push rbp
```

Comment:
So, here's something I just learned... x64 functions must not begin with a single-byte instruction (/hotpatch is always implicitly on, and no NOP is used). PSDK's shared/macamd64.inc has a rex_push_reg macro that explains this (somewhat). MS uses a redundant 0x48 REX prefix to ensure this.
I know we have lots of code failing this requirement, but we should probably follow it going forward.

Reply:
We should probably create that include file and start using it. Seems like the right way to make things portable (eventually) and avoid mistakes.

Reply:
>  x64 functions must not begin with a single-byte instruction [...]

And what's the technical/philosophical reason behind this?

Reply:
I answered that in parentheses right after the part you quoted. Hotpatching needs a two byte instruction (otherwise a running instance of the function might be in the middle of the first, then try to read the second instruction which has been overwritten with the second half of the patched-in jmp). x64 doesn't use "mov edi, edi", it mandates instead that functions must start with a two-byte (or larger) instruction.